http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/PackageSpec.scala

```diff
@@ -21,4 +25,36 @@ class PackageSpec extends AnyFlatSpec with Matchers {
     val input1 = "foo,bar"
     processLabelsToReturn(input1) shouldBe Right(List("foo", "bar"))
   }
+
+  it should "process creatorOnly parameter when role=creator" in {
+    val email = WorkbenchEmail("user1@example.com")
+    val params = Map("role" -> "creator")
+    processCreatorOnlyParameter(email, params, TraceId(UUID.randomUUID())) shouldBe Right(Some(email))
+  }
+
+  it should "process creatorOnly parameter when creator=caller_email" in {
+    val email = WorkbenchEmail("user1@example.com")
+    val params = Map("creator" -> email.value)
+    processCreatorOnlyParameter(email, params, TraceId(UUID.randomUUID())) shouldBe Right(Some(email))
+  }
+
+  it should "process creatorOnly parameter when no role specified" in {
+    val email = WorkbenchEmail("user1@example.com")
+    val params: Map[String, String] = Map.empty
+    processCreatorOnlyParameter(email, params, TraceId(UUID.randomUUID())) shouldBe Right(None)
+  }
+
+  it should "reject creatorOnly parameter when role=not_creator" in {
+    val email = WorkbenchEmail("user1@example.com")
+    val params = Map("role" -> "owner")
+    val res = processCreatorOnlyParameter(email, params, TraceId(UUID.randomUUID()))
+    res.isLeft
```

Comment:
Do we need to check if this is a BadRequestException? since it's typed that way could it ever be anything else? I tried a number of ways to assert that the value contained in the Left is a BadRequestException and couldn't get anything to successfully compile, so either this is unnecessary or I'm bad at Scala (likely)

Reply:
you can do sth like `res shouldBe Left(BadRequestException(???))`